test/integration/controller_test.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"testing"
+	"time"
+
+	"k8s.io/client-go/1.5/kubernetes/fake"
+	"k8s.io/kubernetes/pkg/api/v1"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1alpha1"
+	"github.com/kubernetes-incubator/service-catalog/pkg/brokerapi"
+	fakebrokerapi "github.com/kubernetes-incubator/service-catalog/pkg/brokerapi/fake"
+	"github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset"
+	scinformers "github.com/kubernetes-incubator/service-catalog/pkg/client/informers"
+	informers "github.com/kubernetes-incubator/service-catalog/pkg/client/informers/servicecatalog/v1alpha1"
+	"github.com/kubernetes-incubator/service-catalog/pkg/controller/wip"
+	"github.com/kubernetes-incubator/service-catalog/pkg/registry/servicecatalog/server"
+)
+
+// TestController is a very basic test to start
+//
+// need etcd running
+// start a fresh apiserver for the controller to talk to
+func TestController(t *testing.T) {
+	fakeKubeClient, catalogClient, fakeBrokerCatalog, _, _, testController, _, stopCh := newTestController(t)
+	defer close(stopCh)
+
+	t.Log(fakeKubeClient, catalogClient, fakeBrokerCatalog, testController, stopCh)
+
+	fakeBrokerCatalog.RetCatalog = &brokerapi.Catalog{
+		Services: []*brokerapi.Service{
+			{
+				Name:        "test-service",
+				ID:          "12345",
+				Description: "a test service",
+				Plans: []brokerapi.ServicePlan{
+					{
+						Name:        "test-plan",
+						Free:        true,
+						ID:          "34567",
+						Description: "a test plan",
+					},
+				},
+			},
+		},
+	}
+	name := "test-name"
+	broker := &v1alpha1.Broker{
+		ObjectMeta: v1.ObjectMeta{Name: name},
+		Spec: v1alpha1.BrokerSpec{
+			URL:     "https://example.com",
+			OSBGUID: "OSBGUID field",
+		},
+	}
+	brokerClient := catalogClient.Servicecatalog().Brokers()
+
+	brokerServer, err := brokerClient.Create(broker)
+	if nil != err {
+		t.Fatalf("error creating the broker %q (%q)", broker, err)
+	}
+
+	timeout := time.After(30 * time.Second)
+	tick := time.Tick(500 * time.Millisecond)
+checkStatus:
```

Comment:
this wait til it's done should be in the client directly, but as it isn't, we should attempt to refactor it into something generically available. There may be some better builtin go construct, but this is my view of idiomatic go.